lib/Group/GroupBackend.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * @copyright Copyright (c) 2024 Arawa
+ *
+ * @author 2024 Sébastien Marinier <seb@smarinier.net>
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Workspace\Group;
+
+use OCA\Workspace\Service\Group\ConnectedGroupsService;
+use OCP\Group\Backend\ABackend;
+use OCP\GroupInterface;
+use OCP\IGroupManager;
+use OCP\IUserManager;
+
+class GroupBackend extends ABackend implements GroupInterface {
+
+	private bool $_avoidRecurse_users;
+	private bool $_avoidRecurse_groups;
+
+	/**
+	 * @param IGroupManager $groupManager parent group manager
+	 */
+	public function __construct(protected IGroupManager $groupManager, protected IUserManager $userManager, private ConnectedGroupsService $connectedGroups) {
+		$this->_avoidRecurse_users = $this->_avoidRecurse_groups = false;
+	}
+
+	/**
+	 * is user in group?
+	 * @param string $uid uid of the user
+	 * @param string $gid gid of the group
+	 * @return bool
+	 * @since 4.5.0
+	 *
+	 * Checks whether the user is member of a group or not.
+	 */
+	public function inGroup($uid, $gid) {
+		return true;
+	}
+
+	/**
+	 * Get all groups a user belongs to
+	 * @param string $uid Name of the user
+	 * @return array an array of group names
+	 * @since 4.5.0
+	 *
+	 * This function fetches all groups a user belongs to. It does not check
+	 * if the user exists at all.
+	 */
+	public function getUserGroups($uid) {
```

Comment:
Could you type your parameters, please? :)

Reply:
refused by my syntax linter, as the interface does not